### ISSUE TITLE
Placeholder name format doesn't ignore variables

### ISSUE THREAD

**Issue by bgriffith** [sasstools/sass-lint#625]
Current the `placeholder-name-format` rule will incorrectly flag up variables.
```scss
$var: 'foo';

%#{$var} {
  font-size: 18px;
}
```

Will incorrectly return `Placeholder %#{$var} should be written in lowercase with hyphens`.

Sass 1.5.1, develop branch

**Comment by nicolas-goudry** [comment]
Hello,

I think this issue is not fixed. I manually configured sass-lint, here is [my config](https://gist.github.com/nicolas-goudry/1d6aef89626c750d17eaff307d67ccb0).

I set
```yaml
files:
  include: 'src/**/*.scss'
```
And `sass-lint` is still processing all folders within my project directory… But that's not the point!

The error is happening with `bemDepth.js` because function is trying to split a `null` value. Here is the stack trace:

```
C:\react-poc\node_modules\sass-lint\lib\rules\bem-depth.js:12
  var elements = str.split('__').length;
                    ^

TypeError: Cannot read property 'split' of null
    at bemDepth (C:\react-poc\node_modules\sass-lint\lib\rules\bem-depth.js:12:21)
    at Node.<anonymous> (C:\react-poc\node_modules\sass-lint\lib\rules\bem-depth.js:37:17)
    at C:\react-poc\node_modules\gonzales-pe\lib\gonzales.js:305:54
    at Node.traverse (C:\react-poc\node_modules\gonzales-pe\lib\gonzales.js:279:6)
    at Node.traverse (C:\react-poc\node_modules\gonzales-pe\lib\gonzales.js:284:36)
    at Node.traverse (C:\react-poc\node_modules\gonzales-pe\lib\gonzales.js:284:36)
    at Node.traverse (C:\react-poc\node_modules\gonzales-pe\lib\gonzales.js:284:36)
    at Node.traverseByTypes (C:\react-poc\node_modules\gonzales-pe\lib\gonzales.js:304:11)
    at Object.module.exports.detect (C:\react-poc\node_modules\sass-lint\lib\rules\bem-depth.js:29:9)
    at C:\react-poc\node_modules\sass-lint\index.js:124:27
    at Array.forEach (native)
    at Function.sassLint.lintText (C:\react-poc\node_modules\sass-lint\index.js:123:11)
    at C:\react-poc\node_modules\sass-lint\index.js:222:23
    at Array.forEach (native)
    at Function.sassLint.lintFiles (C:\react-poc\node_modules\sass-lint\index.js:219:12)
    at lint (C:\react-poc\node_modules\sasslint-webpack-plugin\lib\linter.js:18:25)
```

This is happening with a `sass-lint` test file: `sass-lint/tests/sass/placeholder-name-format.sass` and rule is at `line 57`:

```sass
%#{$var}
  content: ''
```

Did I do something wrong or did I found a non resolved bug?

Thanks for this great work anyway, I removed the rule from config for now and it's working as expected!

**Comment by DanPurdy** [comment]
@nicolas-goudry This is a separate issue, this issue is to do with the placeholder name format rule, yours is the bem depth rule.

Could you please open a new issue and include the information that is requested, we can then investigate it for you. Thanks! 👍

**Comment by nicolas-goudry** [comment]
@DanPurdy It's done, thanks! ==> [https://github.com/sasstools/sass-lint/issues/782](https://github.com/sasstools/sass-lint/issues/782)
